cmd/manager/main.go

```diff
@@ -143,14 +151,48 @@ func main() {
 		os.Exit(1)
 	}
 
-	baseStorageURL, err := url.Parse(fmt.Sprintf("%s/catalogs/", httpExternalAddr))
+	switch {
+	case certFile == "" && keyFile == "":
+		listener, err = net.Listen("tcp", catalogServerAddr)
+		if err != nil {
+			setupLog.Error(err, "unable to create HTTP server listener")
+			os.Exit(1)
+		}
+		externalAddr = "http://" + externalAddr
+	case certFile != "" && keyFile != "":
+		tlsFileWatcher := certwatcher.TLSCertificateWatcher{
+			Logger:   ctrl.Log.WithName("tls-file-watcher"),
+			CertFile: certFile,
+			KeyFile:  keyFile,
+		}
+		config := &tls.Config{
+			GetCertificate: tlsFileWatcher.GetCertificate,
+			MinVersion:     tls.VersionTLS13,
+		}
+		listener, err = tls.Listen("tcp", catalogServerAddr, config)
+		if err != nil {
+			setupLog.Error(err, "unable to create HTTPS server listener")
+			os.Exit(1)
+		}
+		if err := mgr.Add(&tlsFileWatcher); err != nil {
+			setupLog.Error(err, "problem adding TLS file watcher to manager")
+			os.Exit(1)
+		}
+		externalAddr = "https://" + externalAddr
+	default:
+		setupLog.Error(nil, "unable to configure TLS certificates, both tls-cert and tls-key arguments are required")
```

Comment:
Nit: "are required" makes it sound like both flags are always required. Maybe soften it a bit to try to clarify that these are both or neither.

```suggestion
		setupLog.Error(nil, "unable to configure TLS certificates, --tls-cert and --tls-key flags are required together")
```